### PR TITLE
chore: fix typos

### PR DESCRIPTION
**Pull request by dundargoc**
Typo fixes and other small pull-requests will be collected here.

### CODE REVIEW

File: src/nvim/api/vim.c [hunk 1]
```diff
@@ -1328,7 +1328,7 @@ Boolean nvim_paste(String data, Boolean crlf, Integer phase, Error *err)
   if (!cancel && !(State & CMDLINE)) {  // Dot-repeat.
     for (size_t i = 0; i < lines.size; i++) {
       String s = lines.items[i].data.string;
-      assert(data.size <= INT_MAX);
+      assert(s.size <= INT_MAX);
```
Comment (clason) [inline src/nvim/api/vim.c:1331]: that doesn't quite look like (just) a typo to me...

Comment (dundargoc) [reply src/nvim/api/vim.c:1331]: https://github.com/neovim/neovim/pull/17752

Comment (clason) [reply src/nvim/api/vim.c:1331]: Yes, I saw that, but that doesn't change my point: functional code changes -- even if minor -- are not "typos" and shouldn't be swept under that rug.

Comment (dundargoc) [reply src/nvim/api/vim.c:1331]: Yes. I'll remove it.

Comment (clason) [reply src/nvim/api/vim.c:1331]: (Not sure the original PR can be reopened, though.)

Comment (dundargoc) [reply src/nvim/api/vim.c]: OK, will create a new one.

Comment (dundargoc) [reply src/nvim/api/vim.c]: Never mind, he reopened his PR.

**Comment by clason** [comment]
Is this girthy enough to merge already, or do you want to wait until shortly before the next release?

**Comment by dundargoc** [comment]
Merge it.

**Comment by clason** [comment]
Done.

**Pull request merged**